File: bms/go/geo/geo_r3_Vector_IsUnit/geo_r3_Vector_IsUnit.go
package test

import (
	"math"
)

func (v geo_r3_Vector) IsUnit() bool {
	const epsilon = 5e-14
	return math.Abs(v.Norm2()-1) <= epsilon
}

func (v geo_r3_Vector) Norm2() float64 { return v.Dot(v) }

func (v geo_r3_Vector) Dot(ov geo_r3_Vector) float64 {
	return float64(v.X*ov.X) + float64(v.Y*ov.Y) + float64(v.Z*ov.Z)
}

type geo_r3_Vector struct {
	X, Y, Z float64
}

File: translations_only/success/go/geo/geo_r3_Vector_IsUnit/claude3/BaseRepair/geo_r3_Vector_IsUnit.rs


const EPSILON: f64 = 5e-14;

struct GeoR3Vector {
    x: f64,
    y: f64,
    z: f64,
}

fn is_unit(v: &GeoR3Vector) -> bool {
    (norm2(&v) - 1.0).abs() <= EPSILON
}

fn norm2(v: &GeoR3Vector) -> f64 {
    dot(v, v)
}

fn dot(v1: &GeoR3Vector, v2: &GeoR3Vector) -> f64 {
    v1.x * v2.x + v1.y * v2.y + v1.z * v2.z
}

